clang/test/CodeGen/ptrauth-cpu-feature.c

```diff
@@ -0,0 +1,59 @@
+// REQUIRES: aarch64-registered-target
+
+// Test that features requiring FEAT_PAuth fail early if the requirement is not met:
+//
+// RUN: not %clang %s -S -o - -target aarch64-linux-gnu -mcpu=cortex-a72 -mbranch-protection=pauthabi 2>&1 | FileCheck %s --check-prefix=FAIL
+// RUN: not %clang %s -S -o - -target aarch64-linux-gnu -mcpu=cortex-a72 -fptrauth-calls              2>&1 | FileCheck %s --check-prefix=FAIL
+// RUN: not %clang %s -S -o - -target aarch64-linux-gnu -mcpu=cortex-a72 -fptrauth-returns            2>&1 | FileCheck %s --check-prefix=FAIL
+// RUN: not %clang %s -S -o - -target aarch64-linux-gnu -mcpu=cortex-a72 -fptrauth-objc-isa           2>&1 | FileCheck %s --check-prefix=FAIL
+//
+// RUN:     %clang %s -S -o - -target aarch64-linux-gnu -mcpu=apple-a12  -mbranch-protection=pauthabi 2>&1 | FileCheck %s --check-prefix=PAUTH
+
+// Test that the following options are still gated on -fptrauth-calls.
```

Comment:
Don't we want to emit a warning or smth in these cases? A user probably meant to enable some PAuth-related feature if one of these options is passed. If we just silently do not actually enable the feature unless `-fptrauth-calls` is here, it might be surprising for the user.

Reply:
Agree, added a warning